### PR TITLE
mergify: Expand the types of PRs that can be merged

### PR DESCRIPTION
**Pull request by russellb**
commit 7fdecb47e1db5af7c3e4a07b1008ea2c82f8bd22
Author: Russell Bryant <russell.bryant@gmail.com>
Date:   Thu Apr 20 15:27:47 2023 -0400

    mergify: Expand the types of PRs that can be merged
    
    This PR defines 3 sets of rules for what CI must pass based on the
    files changed in the PR.
    
    1) If a PR only runs the docs workflow, only check for `markdown-lint`
       passing. This is based on checking the file types changed in the
       PR. The regex must stay in sync with the file types listed in the
       github workflows, so I added a comment across the 3 files as a
       reminder.
    
    2) If a PR has no markdown file changes, it only needs to pass the
       build workflow.
    
    3) If a PR also includes markdown changes, it must pass the build
       workflow and markdown-lint.

    Also add a requirement that the DCO check passes.
    
    Signed-off-by: Russell Bryant <rbryant@redhat.com>

commit 16d0167d3c681d003c14c55e6062ce050f1c243f 
Author: Russell Bryant <russell.bryant@gmail.com>
Date:   Thu Apr 20 17:22:31 2023 -0400

    ci: Drop enforce-labels workflow
    
    The functionality here has been replaced by using mergify to enforce
    merge policy.
    
    Signed-off-by: Russell Bryant <rbryant@redhat.com>

### CODE REVIEW

**Review by vishnoianil — APPROVED** [review]
LGTM, waiting till tomorrow in case other maintainer has any feedback.

**Review by dave-tucker — APPROVED** [review]
LGTM

**Comment by dave-tucker** [comment]
Oh cool, looks like mergify won't merge changes to it's config unless you explicitly allow it 🎉

**Pull request merged**
